pkg/kubelet/cm/cpumanager/state/checkpoint.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"encoding/json"
+	"hash/fnv"
+
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/errors"
+	hashutil "k8s.io/kubernetes/pkg/util/hash"
+)
+
+// cpuManagerCheckpoint struct is used to store cpu/pod assigments in a checkpoint
+type cpuManagerCheckpoint struct {
+	PolicyName string
+	DefaultCPUSet string
+	Entries map[string]string
+	Checksum uint64
+}
+
+// NewCPUManagerCheckpoint returns an instance of Checkpoint
+func NewCPUManagerCheckpoint() checkpointmanager.Checkpoint {
```

Comment:
If you return `cpuManagerCheckpoint` instead of `checkpointmanager.Checkpoint` you might be able to avoid typo conversions later (as `cpuManagerCheckpoint` is always passed as an argument to another function and never returned).